### PR TITLE
Handle columns with double underscores in the name

### PR DESCRIPTION
**Pull request by Limess**
For example 'properties__number_of_sources'.

Previously anything prefixed by 'properties__' was coalesced to 'properties'.

For example this was being generated against our events using segment -> stitch -> Redshift:

```sql
WITH final AS (
  
    
    
select

    coalesce(
                cast("properties__meta__content__count__string" as 
  varchar
),
                cast("properties__meta__frequency__at__string" as 
  varchar
),
                cast("properties__meta__frequency__between__from__string" as 
  varchar
),
                cast("properties__meta__frequency__between__to__string" as 
  varchar
),
                cast("properties__created__string" as 
  varchar
),
                cast("properties__meta__frequency__between__from__bigint" as 
  varchar
),
                cast("properties__meta__frequency__at__bigint" as 
  varchar
),
                cast("properties__meta__frequency__between__to__bigint" as 
  varchar
),
                case
                    when "properties__created__boolean" = true then cast('true' as 
  varchar
)
                    when "properties__created__boolean" = false then cast('false' as 
  varchar
)
                    else null end
                ,
                cast("properties__item__journalistic_quality__double" as 
  varchar
),
                cast("properties__item__journalistic_quality__bigint" as 
  varchar
),
                cast("properties__item__original_id__string" as 
  varchar
),
                cast("properties__sentiment__compound__bigint" as 
  varchar
),
                cast("properties__sentiment__pos__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__neu__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__pos__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__neg__bigint" as 
  varchar
),
                cast("properties__sentiment__neu__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__compound__bigint" as 
  varchar
),
                cast("properties__sentiment__neg__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__compound__double" as 
  varchar
),
                cast("properties__sentiment__neu__double" as 
  varchar
),
                cast("properties__entity__sentiment__pos__double" as 
  varchar
),
                cast("properties__entity__sentiment__neu__double" as 
  varchar
),
                cast("properties__sentiment__compound__double" as 
  varchar
),
                cast("properties__sentiment__pos__double" as 
  varchar
),
                cast("properties__sentiment__neg__double" as 
  varchar
),
                cast("properties__entity__sentiment__neg__double" as 
  varchar
),
                cast("properties__item__original_id__bigint" as 
  varchar
),
                cast("properties__number_of_sources__string" as 
  varchar
),
                cast("properties__number_of_sources__bigint" as 
  varchar
),
                cast("properties__meta__content__count__bigint" as 
  varchar
)) as "properties"

from "analytics"."migrated_app"."track"


)

SELECT * FROM final
```

After this change:

```sql
    coalesce(
                cast("properties__created__string" as 
  varchar
),
                case
                    when "properties__created__boolean" = true then cast('true' as 
  varchar
)
                    when "properties__created__boolean" = false then cast('false' as 
  varchar
)
                    else null end
                ) as "properties__created",
    coalesce(
                cast("properties__entity__sentiment__compound__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__compound__double" as 
  varchar
)) as "properties__entity__sentiment__compound",
    coalesce(
                cast("properties__entity__sentiment__neg__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__neg__double" as 
  varchar
)) as "properties__entity__sentiment__neg",
    coalesce(
                cast("properties__entity__sentiment__neu__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__neu__double" as 
  varchar
)) as "properties__entity__sentiment__neu",
    coalesce(
                cast("properties__entity__sentiment__pos__bigint" as 
  varchar
),
                cast("properties__entity__sentiment__pos__double" as 
  varchar
)) as "properties__entity__sentiment__pos",
    coalesce(
                cast("properties__item__journalistic_quality__double" as 
  varchar
),
                cast("properties__item__journalistic_quality__bigint" as 
  varchar
)) as "properties__item__journalistic_quality",
    coalesce(
                cast("properties__item__original_id__string" as 
  varchar
),
                cast("properties__item__original_id__bigint" as 
  varchar
)) as "properties__item__original_id",
    coalesce(
                cast("properties__meta__content__count__string" as 
  varchar
),
                cast("properties__meta__content__count__bigint" as 
  varchar
)) as "properties__meta__content__count",
    coalesce(
                cast("properties__meta__frequency__at__string" as 
  varchar
),
                cast("properties__meta__frequency__at__bigint" as 
  varchar
)) as "properties__meta__frequency__at",
    coalesce(
                cast("properties__meta__frequency__between__from__string" as 
  varchar
),
                cast("properties__meta__frequency__between__from__bigint" as 
  varchar
)) as "properties__meta__frequency__between__from",
    coalesce(
                cast("properties__meta__frequency__between__to__string" as 
  varchar
),
                cast("properties__meta__frequency__between__to__bigint" as 
  varchar
)) as "properties__meta__frequency__between__to",
    coalesce(
                cast("properties__number_of_sources__string" as 
  varchar
),
                cast("properties__number_of_sources__bigint" as 
  varchar
)) as "properties__number_of_sources",
    coalesce(
                cast("properties__sentiment__compound__bigint" as 
  varchar
),
                cast("properties__sentiment__compound__double" as 
  varchar
)) as "properties__sentiment__compound",
    coalesce(
                cast("properties__sentiment__neg__bigint" as 
  varchar
),
                cast("properties__sentiment__neg__double" as 
  varchar
)) as "properties__sentiment__neg",
    coalesce(
                cast("properties__sentiment__neu__bigint" as 
  varchar
),
                cast("properties__sentiment__neu__double" as 
  varchar
)) as "properties__sentiment__neu",
    coalesce(
                cast("properties__sentiment__pos__bigint" as 
  varchar
),
                cast("properties__sentiment__pos__double" as 
  varchar
)) as "properties__sentiment__pos"

from "analytics"."migrated_app"."track"

```

### CODE REVIEW

**Review by jtcohen6 — APPROVED** [review]
@Limess Thanks for the PR! This fix looks great to me, and really appreciate the update to integration tests.

Since Stitch uses double underscores in column names to signify flattened or unnested JSON fields, I'm glad the macro will now be smarter handling those.

File: integration_tests/data/data_coalesce_fields.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-id,numeral,numeral__st,numeral__bl,version__string,version__double,__red_herring
-1,1,,,pi,,not
-2,,one,,,3.14,this
-3,,,TRUE,,,one
+id,numeral,numeral__st,numeral__bl,version__string,version__double,properties__numeral__st,properties__numeral__bl,__red_herring
+1,1,,,pi,,1,not
+2,,one,,,3.14,one,this
+3,,,TRUE,,,,,one
```
Comment (jtcohen6) [inline integration_tests/data/data_coalesce_fields.csv]: I just ran and failed integration tests locally. I think these columns may just be slightly misaligned.
```suggestion
id,numeral,numeral__st,numeral__bl,version__string,version__double,properties__numeral__st,properties__numeral__it,properties__numeral__bl,__red_herring
1,1,,,pi,,,1,,not
2,,one,,,3.14,one,,,this
3,,,TRUE,,,,,true,one
```

**Comment by jtcohen6** [comment]
@Limess Looks great, thanks for updating! I'll merge this.

**Comment by Limess** [comment]
Thanks for the quick turnaround! Would it be possible to publish the latest release to DBT hub (https://hub.getdbt.com/fishtown-analytics/stitch_utils/latest/)?

**Comment by jtcohen6** [comment]
Release 0.1.6 should be deployed on the hub site now!

**Pull request merged**